src/service/private.overseas.entity.details.ts

```diff
@@ -18,15 +18,40 @@ export const getPrivateOeDetails = async (
     overseasEntityId,
   );
 
+  checkErrorResponse(req, response, overseasEntityId, transactionId);
+
+  return response.resource;
+};
+
+export const getBeneficialOwnerPrivateData = async (
+  req: Request,
+  transactionId: string,
+  overseasEntityId: string
+): Promise<BeneficialOwnersPrivateData | undefined> => {
+  const response = await makeApiCallWithRetry(
+    "overseasEntity",
+    "getBeneficialOwnerPrivateData",
+    req,
+    req.session as Session,
+    transactionId,
+    overseasEntityId
+  );
+
+  checkErrorResponse(req, response, overseasEntityId, transactionId);
+
+  return response.resource;
+};
+
+const checkErrorResponse = (req: Request, response, overseasEntityId: string, transactionId: string) => {
```

Comment:
maybe pass in what kind of data was being fetched and include that string in the errorMsg

Reply:
could be optional arg